### PR TITLE
[opaque pointers 5] Use AddressInfo in LoadInst

### PR DESCRIPTION
**Pull request by aneshlya**
This PR switches `StoreInst` to use `AddressInfo` class.
Also eliminates usage of `getPointerElementType` in easy cases.

### CODE REVIEW

File: src/module.cpp [hunk 1]
```diff
@@ -2782,10 +2779,12 @@ static bool lCompatibleTypes(llvm::Type *Ty1, llvm::Type *Ty2) {
             break;
 
         case llvm::ArrayType::PointerTyID:
+// In opaque mode pointers are compatible
```
Comment (dbabokin) [inline src/module.cpp:2782]: Is it intended behavior to have fall-though in Opaque Mode?

Comment (dbabokin) [reply src/module.cpp:2782]: I think it was assumes to `return true;` in else case of ifndef.
It also seems to affect how ISPC reports warnings about pointers to varying types in global variables.

Comment (aneshlya) [reply src/module.cpp]: The intention was to have `ifndef ` before `case llvm::ArrayType::PointerTyID`, not after. But anyway I reworked it in the right way.

**Review by dbabokin — APPROVED** [review]
Looks good now!

**Pull request merged**